AzureStorageBlob/Source/DataStructures/StorageStreamDownloader.swift

```diff
@@ -0,0 +1,435 @@
+        let modifiedAccessConditions = options.modifiedAccessConditions
+        let cpk = options.cpk
+
+        headerParams["x-ms-range"] = "bytes=\(startRange)-\(endRange)"
+        if let rangeGetContentMD5 = options.range?.calculateMD5 { headerParams["x-ms-range-get-content-md5"] = String(rangeGetContentMD5) }
+        if let rangeGetContentCRC64 = options.range?.calculateCRC64 { headerParams["x-ms-range-get-content-crc64"] = String(rangeGetContentCRC64) }
+
+        if let requestId = requestId { headerParams["x-ms-client-request-id"] = requestId }
+        if let leaseId = leaseAccessConditions?.leaseId { headerParams["x-ms-lease-id"] = leaseId }
+        if let encryptionKey = cpk?.value { headerParams["x-ms-encryption-key"] = String(data: encryptionKey, encoding: .utf8) }
+        if let encryptionKeySHA256 = cpk?.hash { headerParams["x-ms-encryption-key-sha256"] = encryptionKeySHA256 }
+        if let encryptionAlgorithm = cpk?.algorithm { headerParams["x-ms-encryption-algorithm"] = encryptionAlgorithm }
+        if let ifModifiedSince = modifiedAccessConditions?.ifModifiedSince { headerParams[.ifModifiedSince] = ifModifiedSince.rfc1123Format }
+        if let ifUnmodifiedSince = modifiedAccessConditions?.ifUnmodifiedSince { headerParams[.ifUnmodifiedSince] = ifUnmodifiedSince.rfc1123Format }
+        if let ifMatch = modifiedAccessConditions?.ifMatch { headerParams[.ifMatch] = ifMatch }
+        if let ifNoneMatch = modifiedAccessConditions?.ifNoneMatch { headerParams[.ifNoneMatch] = ifNoneMatch }
+
+        // Construct and send request
+        let request = client.request(method: HttpMethod.GET,
+                                   url: url,
+                                   queryParams: queryParams,
+                                   headerParams: headerParams)
+        let context: [String: AnyObject] = [
+            ContextKey.allowedStatusCodes.rawValue: [200, 206] as AnyObject,
+        ]
+        client.run(request: request, context: context) { result, httpResponse in
+            switch result {
+            case let .failure(error):
+                completion(.failure(error), httpResponse)
+            case let .success(data):
+                guard let data = data else {
+                    completion(.failure(AzureError.general("Blob unexpectedly contained no data.")), httpResponse)
+                    return
+                }
+                let headers = httpResponse.headers
+                if let contentMD5 = headers["Content-MD5"] {
+                    let dataHash = data.md5
+                    guard contentMD5 == dataHash else {
+                        let error = AzureError.general("Block MD5 \(dataHash) did not match \(contentMD5).")
+                        completion(.failure(error), httpResponse)
+                        return
+                    }
+                }
+                if let contentCRC64 = headers["Content-CRC64"] {
+                    let dataHash = data.crc64
```

Comment:
If we get here, this is always going to fail right? Because we don't have an implementation of crc64 yet. Should we comment this block out?

Reply:
I think the TODO is okay for now. Prior to releasing Preview 1, we would resolve all of the TODOs, either by providing the implementation or removing the capability. Once these PRs are merged, I'll create issues for each of these outstanding TODOs.